contracts/protocol/libraries/logic/GenericLogic.sol

```diff
@@ -129,56 +133,83 @@ library GenericLogic {
                 vars.liquidationThreshold,
                 vars.liquidationBonus,
                 vars.decimals,
-
+                ,
+                vars.dynamicConfigs
             ) = currentReserve.configuration.getParams();
 
             unchecked {
                 vars.assetUnit = 10**vars.decimals;
             }
 
+            vars.assetPrice = IPriceOracleGetter(params.oracle).getAssetPrice(
+                vars.currentReserveAddress
+            );
+
             if (
-                vars.liquidationThreshold != 0 &&
+                (vars.liquidationThreshold != 0 || vars.dynamicConfigs) &&
                 params.userConfig.isUsingAsCollateral(vars.i)
             ) {
-                (
-                    vars.userBalanceInBaseCurrency,
-                    vars.assetPrice
-                ) = _getUserBalanceInBaseCurrency(
-                    params.user,
-                    currentReserve,
-                    params.oracle,
-                    vars.assetUnit,
-                    vars.currentReserveAddress
-                );
-
-                vars.totalCollateralInBaseCurrency += vars
-                    .userBalanceInBaseCurrency;
-
-                if (vars.ltv != 0) {
-                    vars.avgLtv += vars.userBalanceInBaseCurrency * vars.ltv;
-                } else {
-                    vars.hasZeroLtvCollateral = true;
-                }
-
-                vars.liquidationThreshold =
-                    vars.userBalanceInBaseCurrency *
-                    (vars.liquidationThreshold);
-
-                vars.avgLiquidationThreshold += vars.liquidationThreshold;
-
                 if (vars.assetType == DataTypes.AssetType.ERC721) {
```

Comment:
May be we should add a AssetType for UniswapV3 NFT from the very beginning, like `ERC721_UNIV3`. then we can have a more straightforward if else logic like below:
```
if (vars.assetType == DataTypes.AssetType.ERC721_UNIV3) {

} else if (vars.assetType == DataTypes.AssetType.ERC721) {

} else {

}
```

Reply:
the problem with this is that our code won't be as flexible and scalable